extension/src/popup/views/__tests__/SignTransaction.test.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { render, waitFor, screen } from "@testing-library/react";
+import * as createStellarIdenticon from "stellar-identicon-js";
+
+import * as Stellar from "helpers/stellar";
+
+import { SignTransaction } from "../SignTransaction";
+
+import { Wrapper } from "../../__testHelpers__";
+
+jest.mock("stellar-identicon-js");
+
+const defaultSettingsState = {
+  networkDetails: {
+    isTestnet: false,
+    network: "",
+    networkName: "",
+    otherNetworkName: "",
+    networkUrl: "",
+    networkPassphrase: "foo",
+  },
+};
+
+const defaultTransactionInfo = {
+  accountToSign: "",
+  transaction: { _networkPassphrase: "foo" },
+  transactionXdr: "",
+  domain: "",
+  domainTitle: "",
+  isHttpsDomain: true,
+  operations: [],
+  operationTypes: [],
+  isDomainListedAllowed: true,
+  flaggedKeys: { test: { tags: [""] } },
+};
+
+describe("SignTransactions", () => {
+  beforeEach(() => {
+    const mockCanvas = document.createElement("canvas");
+    jest.spyOn(createStellarIdenticon, "default").mockReturnValue(mockCanvas);
+  });
+  it("renders", async () => {
+    jest
+      .spyOn(Stellar, "getTransactionInfo")
+      .mockImplementation(() => defaultTransactionInfo);
+
+    render(
+      <Wrapper
+        state={{
+          settings: defaultSettingsState,
+        }}
+      >
+        <SignTransaction />
+      </Wrapper>,
+    );
+    await waitFor(() => screen.getByTestId("SignTransaction"));
+    expect(screen.getByTestId("SignTransaction")).toBeDefined();
+  });
+  it("shows non-https domain error", async () => {
```

Comment:
there's a lot of other use cases to cover here, but this took me a long time to get this whole scaffold up and running so in the interest of time, just adding the pertinent cases for this PR